### PR TITLE
Flask

### LINKED ISSUE #98: Add network selector widget
Networks can be selected by http request. There should be a selector which features all possible networks and switches between them.

See: http://biobits.org/bokeh-flask.html

### PR DESCRIPTION
**Pull request by JimMadge**
Closes #98 

@edwardchalstrey1 The network selection drop-down has returned via flask.

To do this the entire bokeh app has to be contained within a single function. This has something like an entrypoint as http requests can be passed from flask to the bokeh document. Normally I'd think that was a good thing, but it does mean nested functions have returned. However, in this case the structure is much more logical with all of the initialisation parts together. Essentially, the old app has been indented and put in a ```def``` block.

Another advantage of this is that it should be quite straightforward to create a file upload widget in flask. When this happens an argument could be passed to the bokeh document that prompts it to use the uploaded data rather than one of the example networks. This might be worth considering.

### CODE REVIEW

**Comment by edwardchalstrey1** [comment]
@JimMadge can you update the README and Dockerfile with the new command for running the app (assuming it's no longer `bokeh serve water` )?

**Comment by JimMadge** [comment]
> @JimMadge can you update the README and Dockerfile with the new command for running the app (assuming it's no longer `bokeh serve water` )?

@edwardchalstrey1 Good catch. I've updated the README. There is no Dockerfile on this branch, but I will pull in dev and update it (and sort any conflicts).

File: README.md [hunk 1]
```diff
@@ -31,20 +31,33 @@ The data for contamination spread scenarios is pre-computed and coupled with inf
...
+gunicorn -w 4 main:app
+```
 
 ### Docker version
 
 1. Pull from Docker Hub: `docker pull turinginst/chance-water`
-2. Run the container `docker run -p 5006:5006 turinginst/chance-water`
-3. Open http://localhost:5006/water in a browser
+2. Run the container `docker run -p 8000:8000 turinginst/chance-water`
+3. Open http://localhost:8000 in a browser
```
Comment (edwardchalstrey1) [inline README.md]: I think might be important to specify this step for the local version as well. The terminal suggests to go to http://127.0.0.1:8000/ which doesn't work properly, but http://localhost:8000 does

Comment (JimMadge) [reply README.md]: On your system? 127.0.0.1 and localhost *should* be the same.

I agree it should also be mentioned with the local instructions too.

Comment (edwardchalstrey1) [reply README.md]: Yes on my laptop they were different for some reason, not sure why that would be

**Pull request merged**